addons/website_cf_turnstile/static/src/js/turnstile.js

```diff
@@ -19,15 +19,29 @@ publicWidget.registry.s_website_form.include({
                          data-response-field-name="turnstile_captcha"
                          data-sitekey="${session.turnstile_site_key}"
                          data-error-callback="throwTurnstileError"
+                         data-before-interactive-callback="TurnstileBeforeInteractive"
+                         data-after-interactive-callback="TurnstileAfterInteractive"
                     ></div>
                     <script class="s_turnstile">
-                        // Rethrow the error, or we only will catch a "Script error" without any info 
+                        // Rethrow the error, or we only will catch a "Script error" without any info
                         // because of the script api.js originating from a different domain.
                         function throwTurnstileError(code) {
                             const error = new Error("Turnstile Error");
                             error.code = code;
                             throw error;
                         }
+                        function TurnstileBeforeInteractive() {
+                            const btn = document.querySelector('.s_website_form_send, .o_website_form_send');
+                            if (btn && !btn.classList.contains('disabled')) {
+                                btn.classList.add('disabled', 'cf_form_disabled');
+                            }
+                        }
+                        function TurnstileAfterInteractive() {
+                            const btn = document.querySelector('.s_website_form_send, .o_website_form_send');
```

Comment:
Is it possible to link it to a specific form ?
I guess this is fine: if the challenge is completed in any form, the submit can become available is all forms of the page.

Reply:
Good catch, a little bit overkill probably, but it is done.